infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -438,6 +444,10 @@ private void finishAllStartupTracesAsFailures(Throwable reason) {
                   serversAndProbesFuture.completeExceptionally(iex);
                 }
                 serversAndProbesFuture.complete(null);
+
+                if (tracingSpan != null) {
```

Comment:
General topic - not about this PR changes:
Wondering if we need to add some utils to tracing framework since we have a lot of places with the same code snippet pattern - if span is present finish it. Maybe we should have some helper class that is created and finished nevertheless the tracer is null or not. This would simplify code a bit.

Reply:
I've reworked it a bit to have a utility class that simplifies these steps.